pyresample/kd_tree.py

```diff
@@ -1061,7 +1040,9 @@ def get_neighbour_info(self):
                             (target_lats <= 90) & (target_lats >= -90))
 
         index_arr, distance_arr = self._query_resample_kdtree(
-            resample_kdtree, target_lons, target_lats, valid_output_idx)
+            resample_kdtree, target_lons, target_lats, valid_output_idx,
+            # None)
+            mask.data.ravel()[valid_input_idx.ravel()])
```

Comment:
in satpy, `get_neighbour_info` is called only once. So this query call should probably be moved to `get_pixels_from_neighbour_info`

Reply:
You mean `get_sample_from_neighbour_info`? But then that changes the interface and changes what gets cached (if `mask` is None).